src/services/request-service.ts

```diff
@@ -0,0 +1,75 @@
+import Promise = require('bluebird');
+import r = require('rethinkdb');
+import uuid = require('uuid');
+import EmailService = require('./email-service');
+import nodemailer = require('nodemailer');
+import q = require('../dbutils/query');
+import models = require('../models/models');
+import errors = require('../errors/errors');
+import config = require('../config');
+
+
+module RequestService {
```

Comment:
Maybe we should put another branch inside `insert`? I'm looking at this issue comment: https://github.com/rethinkdb/rethinkdb/issues/3235#issuecomment-60283038

Can you give it a try and see if it still create duplicates? If it works, I need to update all our update queries.